src/libs/actions/User.ts

```diff
@@ -469,6 +471,73 @@ function triggerNotifications(onyxUpdates: OnyxServerUpdate[]) {
     });
 }
 
+function playSoundForMessageType(pushJSON: OnyxServerUpdate[]) {
+    try {
+        const reportActionsOnly = pushJSON.filter((update) => update.key.includes('reportActions_'));
+        const flatten = reportActionsOnly.flatMap((update) => {
+            const value = update.value as OnyxCollection<ReportAction>;
+
+            if (!value) {
+                return [];
+            }
+
+            return Object.values(value);
+        }) as ReportAction[];
+        const types = flatten.map((data) => data?.originalMessage).filter(Boolean) as OriginalMessage[];
```

Comment:
I am wondering if we can write this as `const types: OriginalMessage[]` and whether typescript lets us do that?

Also NAB but maybe move types down after the first for loop, because there is the chance that we calculate types, although it won't be used if the first for loop already returns.

Reply:
> I am wondering if we can write this as const types: OriginalMessage[] and whether typescript lets us do that?

Nope

<img width="765" alt="image" src="https://github.com/Expensify/App/assets/22820318/f5e9b159-131a-4049-a711-793b0597931f">

> Also NAB but maybe move types down after the first for loop, because there is the chance that we calculate types, although it won't be used if the first for loop already returns.

That's a good note!